lib/tests/test_rewrite.rs

```diff
@@ -1077,6 +1078,56 @@ fn test_rebase_descendants_update_bookmark_after_abandon() {
     );
 }
 
+#[test]
+fn test_rebase_descendants_delete_bookmark_after_abandon() {
```

Comment:
Would be good to also have tests showing what happens when one or all targets of a conflicted bookmark get abandoned. It's not obvious what we should do. Let's say the bookmark target is a conflict like `A+(B-C)+(D-E)`. Perhaps we should apply a diff that removes the target, i.e. by replacing it by `absent()`. So if you abandon the `A` commit, the new conflict is `A+(B-C)+(D-E)+(∅-A)= ∅+(B-C)+(D-E)`. What do you think?

Reply:
Added (and updated) tests. I also reverted the doc change since bookmark conflicts can't always be resolved by `jj abandon`.